src/components/Button.tsx

```diff
@@ -0,0 +1,27 @@
+import cn from 'classnames';
+import { MouseEventHandler, ReactNode } from 'react';
+
+type Button = {
+  type: "button" | "submit" | "reset" | undefined;
+  className: string;
+  children: ReactNode;
+  onClick: MouseEventHandler | undefined;
```

Comment:
Lets add an optional field syntax here pease.

```suggestion
  onClick?: MouseEventHandler;
```